### PR TITLE
Fix Foxx security

### PR DESCRIPTION
**Pull request by pluma**
This addresses the bugs introduced in PR #8845.

Specifically:

* when `foxx.api` is set to `false`, aardvark no longer shows an "internal server error" page; instead aardvark works correctly as expected but with the _Services_ tab removed and the Foxx related frontend routes redirect to the dashboard (like the cluster related routes do when cluster is disabled)
* when `foxx.api` is set to `false`, the Foxx API routes no longer respond with a generic 404 error; instead they produce a new specific error `3099` that tools like Foxx CLI can use to show a relevant error message
* when `foxx.store` is set to `false`, the "install service" and "upgrade/replace service" pages no longer show an error message; instead the _Store_ tab is removed and the _Upload_ tab is selected by default

This also includes a workaround for a long-standing bug of the subnav not showing up if the navigation is initialised after the content is rendered.

I'd like a review from @hkernbach for the frontend changes and from @ObiWahn for the backend changes. Because of the scope of this PR I'd also like a thumbs up from @mchacki before merging this.

Additionally /cc FYI @dothebart @jsteemann who worked on the original PR with @ObiWahn

### CODE REVIEW

**Comment by pluma** [comment]
Please read the commit notes for explanations. I've tried to do this piece-by-piece and explain the motivations behind each changeset.

File: js/apps/system/_admin/aardvark/APP/frontend/js/views/installServiceView.js [hunk 1]
```diff
@@ -89,15 +89,10 @@
...
-      if (frontendConfig.disableFoxxStore) {
-        $(this.el).html('<div>Foxx store is disabled via server configuration.</div>');
-        return this;
-      } else {
-        // if repo not fetched yet, wait
-        $(this.el).html(this.template.render({
-          services: this.collection
-        }));
-      }
+      // if repo not fetched yet, wait
```
Comment (hkernbach) [inline js/apps/system/_admin/aardvark/APP/frontend/js/views/installServiceView.js]: do not know the origin of that comment here, but does it make sense?

**Comment by pluma** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/4506/

File: js/common/bootstrap/errors.js [hunk 2]
```diff
@@ -366,43 +367,5 @@
 
   // For compatibility with <= 3.3
   internal.errors.ERROR_ARANGO_COLLECTION_NOT_FOUND = internal.errors.ERROR_ARANGO_DATA_SOURCE_NOT_FOUND;
-
-  //arg1 can be a code or error object
-  internal.throwArangoError = function (arg1, message, httpCode) {
```
Comment (pluma) [inline js/common/bootstrap/errors.js]: I'm not sure where this came from. I rebuilt with maintainer mode and this code vanished.

**Comment by fceller** [comment]
please do a full review. I have fixed some tests

**Pull request merged**